response/max_age.go

```diff
@@ -2,12 +2,41 @@ package response
 
 import (
 	"context"
+	"regexp"
+	"strings"
 
+	"github.com/ONSdigital/dp-legacy-cache-proxy/config"
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
-func maxAge(ctx context.Context, path string) int {
-	log.Info(ctx, "Calculating max-age for "+path)
-	// TODO To be implemented as part of DIS-411
-	return 42
+func maxAge(ctx context.Context, uri string, cfg *config.Config) int {
+	log.Info(ctx, "Calculating max-age for "+uri)
+
+	if isVersionedURI(uri) || isOnsURI(uri) || isLegacyAssetURI(uri) {
```

Comment:
hopefully, a future PR can re-order these thus

```suggestion
	if isLegacyAssetURI(uri) || isOnsURI(uri) || isVersionedURI(uri) {
```

because I think the most common should be first and the most expensive (regex) should be last

though, I've not got the data to back me up on this, so I may be wrong! 😊